client/boot/common.js

```diff
@@ -427,13 +427,32 @@ const boot = ( currentUser, registerRoutes ) => {
 			registerRoutes();
 		}
 
+		const isDesktop = config.isEnabled( 'desktop' );
+		const loggedIn = currentUser.get() !== false;
+
 		// Render initial `<Layout>` for non-isomorphic sections.
 		// Isomorphic sections will take care of rendering their `<Layout>` themselves.
-		if ( ! document.getElementById( 'primary' ) ) {
+		if ( ! isDesktop && ! document.getElementById( 'primary' ) ) {
 			renderLayout( reduxStore );
+
+			page.start( { decodeURLComponents: false } );
 		}
 
-		page.start( { decodeURLComponents: false } );
+		if ( isDesktop && loggedIn ) {
```

Comment:
Taking another look at this, I'm actually a little surprised this refactor works. 

Unless I'm interpreting this incorrectly. this will make it so if (1) we are in a Desktop context, and (2) the user is logged in, then (3) rendering will be blocked on the `cookie-auth-complete` event.

However, the `cookie-auth-complete` event is only fired once (on the window's creation). So I'm confused why this would work after the fact, e.g. when a user navigates to a different page. Wouldn't this check be triggered by each page navigation event, or am I not understanding how this client logic works?

cc: @belcherj

Reply:
My guess is this `render` logic is fired only on application boot. From that point on, the client takes over rendering. Which is the only way the above would make sense.

If this logic was fired for each page navigation, then this probably wouldn't work as expected.